slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Importer.java

```diff
@@ -0,0 +1,188 @@
+      final SignedBlocksDao signedBlocksDao,
+      final SignedAttestationsDao signedAttestationsDao,
+      final ObjectMapper mapper) {
+    this.jdbi = jdbi;
+    this.validatorsDao = validatorsDao;
+    this.signedBlocksDao = signedBlocksDao;
+    this.signedAttestationsDao = signedAttestationsDao;
+    this.mapper = mapper;
+  }
+
+  public void importData(final InputStream input) throws IOException {
+
+    try (final JsonParser jsonParser = mapper.getFactory().createParser(input)) {
+      final ObjectNode rootNode = mapper.readTree(jsonParser);
+
+      final JsonNode metadataJsonNode = rootNode.get("metadata");
+      final Metadata metadata = mapper.treeToValue(metadataJsonNode, Metadata.class);
+
+      if (metadata.getFormatVersion() != FORMAT_VERSION) {
+        throw new IllegalStateException(
+            "Expecting an interchange_format_version of " + FORMAT_VERSION);
+      }
+
+      final ArrayNode dataNode = rootNode.withArray("data");
+      jdbi.useTransaction(
+          h -> {
+            for (int i = 0; i < dataNode.size(); i++) {
+              try {
+                final JsonNode validatorNode = dataNode.get(i);
+                parseValidator(h, validatorNode);
+              } catch (final IllegalArgumentException e) {
+                LOG.error("Failed to parse validator {}, due to {}", i, e.getMessage());
+                throw e;
+              }
+            }
+          });
+    }
+  }
+
+  private void parseValidator(final Handle h, final JsonNode node) throws JsonProcessingException {
+    if (node.isArray()) {
+      throw new IllegalStateException("Element of 'data' was not an object");
+    }
+    final ObjectNode parentNode = (ObjectNode) node;
+    final String pubKey = parentNode.required("pubkey").textValue();
+    final Validator validator = validatorsDao.insertIfNotExist(h, Bytes.fromHexString(pubKey));
+
+    final ArrayNode signedBlocksNode = parentNode.withArray("signed_blocks");
+    importBlocks(h, validator, signedBlocksNode);
+
+    final ArrayNode signedAttestationNode = parentNode.withArray("signed_attestations");
+    importAttestations(h, validator, signedAttestationNode);
+  }
+
+  private void importBlocks(
+      final Handle h, final Validator validator, final ArrayNode signedBlocksNode)
+      throws JsonProcessingException {
+    for (int i = 0; i < signedBlocksNode.size(); i++) {
+      final SignedBlock jsonBlock = mapper.treeToValue(signedBlocksNode.get(i), SignedBlock.class);
+      final BlockValidator blockValidator =
+          new BlockValidator(
+              h,
+              jsonBlock.getSigningRoot(),
+              jsonBlock.getSlot(),
+              validator.getId(),
+              signedBlocksDao);
+
+      if (blockValidator.existsInDatabase()) {
+        LOG.debug(
+            "Block {} for validator {} already exists in database, not imported",
+            i,
+            validator.getPublicKey());
+        continue; // DO NOT duplicate import.
```

Comment:
Don't love the continue. Only thought, perhaps moving the body of block import out you could do an early return. But probably ends pretty similar.

Reply:
we could make a "bool canImportBlock()" - which prints out requisite error msgs, and then returns if it can be imported or not.